tools/document-pii/src/item_collector.rs

```diff
@@ -0,0 +1,325 @@
+        self.all_types
+            .insert(enum_name, EnumOrStruct::Enum(node.clone()));
+    }
+
+    fn visit_item_use(&mut self, i: &'ast syn::ItemUse) {
+        let use_statements = usetree_to_paths(&i.tree, &self.module_path)
+            .iter()
+            .filter(|s| s.contains("relay"))
+            .cloned()
+            .collect();
+
+        self.insert_scoped_paths(use_statements);
+    }
+}
+
+fn normalize_type_path(mut path: String, crate_root: &str, module_path: &str) -> String {
+    path = path
+        .replace(' ', "")
+        .replace('-', "_")
+        .replace("crate::", &format!("{}::", crate_root));
+
+    if path.contains("super::") {
+        let parent_module = {
+            let mut parts = module_path.split("::").collect::<Vec<_>>();
+            parts.pop();
+            parts.join("::")
+        };
+        path = path.replace("super::", &parent_module);
+    }
+    path
+}
+
+/// First flattens the UseTree and then normalizing the paths.
+fn usetree_to_paths(use_tree: &UseTree, module_path: &str) -> Vec<String> {
+    let crate_root = module_path.split_once("::").map_or(module_path, |s| s.0);
+    let paths = flatten_use_tree(
+        syn::Path {
+            leading_colon: None,
+            segments: Punctuated::new(),
+        },
+        use_tree,
+    );
+
+    paths
+        .into_iter()
+        .map(|path| normalize_type_path(path, crate_root, module_path))
+        .collect()
+}
+
+/// Flattens a usetree. For example: use relay_general::protocol::{Foo, Bar,
+/// Baz} into [relay_general::protocol::Foo, relay_general::protocol::Bar, relay_general::protocol::Baz]
+fn flatten_use_tree(mut leading_path: syn::Path, use_tree: &UseTree) -> Vec<String> {
+    match use_tree {
+        UseTree::Path(use_path) => {
+            leading_path.segments.push(use_path.ident.clone().into());
+            flatten_use_tree(leading_path, &use_path.tree)
+        }
+        UseTree::Name(use_name) => {
+            leading_path.segments.push(use_name.ident.clone().into());
+            vec![quote::quote!(#leading_path).to_string()]
+        }
+        UseTree::Group(use_group) => {
+            let mut paths = Vec::new();
+            for item in &use_group.items {
+                paths.extend(flatten_use_tree(leading_path.clone(), item));
+            }
+            paths
+        }
+
+        UseTree::Rename(use_rename) => {
+            leading_path.segments.push(use_rename.rename.clone().into());
+            vec![quote::quote!(#leading_path).to_string()]
+        }
+        // Currently this script can't handle glob imports, which, we shouldn't use anyway.
+        UseTree::Glob(_) => vec![quote::quote!(#leading_path).to_string()],
+    }
+}
+
+fn crate_name_from_file(file_path: &Path) -> anyhow::Result<String> {
+    let file_str = file_path
+        .as_os_str()
+        .to_str()
+        .ok_or_else(|| anyhow!("Invalid file path: {}", file_path.display()))?;
```

Comment:
what if we use `to_string_lossy()` instead? Does it change much? 

```suggestion
        let file_str = file_path.as_os_str().to_string_lossy();
```

Reply:
that works well, thanks!